tests/Craftwork/IdObfuscator/CharacterSetTest.php

```diff
@@ -0,0 +1,28 @@
+<?php
+namespace Craftwork\IdObfuscator;
+
+use Craftwork\IdObfuscator\Exception\DuplicateCharacterException;
+use Craftwork\IdObfuscator\Exception\InvalidCharacterSetException;
+use Craftwork\IdObfuscator\Exception\InvalidCharacterSetLengthException;
+use PHPUnit\Framework\TestCase;
+
+class CharacterSetTest extends TestCase
+{
+    public function testThrowsInvalidCharacterSetLengthExceptionIfCustomCharacterSetTooShort()
+    {
+        $this->expectException(InvalidCharacterSetLengthException::class);
+        CharacterSet::ofCustomCharacters('a');
+    }
+
+    public function testThrowsDuplicateCharacterExceptionIfCustomCharacterSetContainsDuplicateCharacters()
+    {
+        $this->expectException(DuplicateCharacterException::class);
+        CharacterSet::ofCustomCharacters('abcda');
+    }
+
+    public function testThrowsInvalidCharacterSetExceptionIfCustomCharacterSetContainsNonAscii()
+    {
+        $this->expectException(InvalidCharacterSetException::class);
+        CharacterSet::ofCustomCharacters('ß');
```

Comment:
Could you add one for something like chr(128) . chr(129) as well? That way we're also tackling extended ascii. Also 'ß' on its own is also a single character (but two bytes) so it isn't clear what exception you would expect from it.